portfolio/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -0,0 +1,80 @@
+import java.util.Iterator;
+
+public final class FindMeetingQuery {
+  public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
+      long duration = request.getDuration();
+      Collection<TimeRange> available = Arrays.asList(TimeRange.WHOLE_DAY);
+
+      //Create list of relevant events
+      for (Iterator<Event> eventIterator = events.iterator(); eventIterator.hasNext();) {
+          Event event = eventIterator.next();
+
+          //Needs to check if at least one person in attendees is in the event
+          boolean overlappingAttendees = false;
+          Iterator reqAttendees = request.getAttendees().iterator();
+         
+          while (reqAttendees.hasNext() && !overlappingAttendees){
+            if (event.getAttendees().contains(reqAttendees.next())){
+              overlappingAttendees = true;
+            }
+          }
+            
+          // Removes that time slot from the availability 
+          if (overlappingAttendees) {
+              Iterator<TimeRange> availableIterator = available.iterator();
+              TimeRange eventTime = event.getWhen();
+              Collection<TimeRange> newAvailability = new ArrayList<TimeRange>();
+              
+               while (availableIterator.hasNext()){
+                   TimeRange availableTime = availableIterator.next();
+                   // Contains case
+                   if(availableTime.end() > eventTime.end() && availableTime.start() <= eventTime.start()){
+                       newAvailability.add(TimeRange.fromStartDuration(availableTime.start(), eventTime.start() - availableTime.start()));
+                       newAvailability.add(TimeRange.fromStartDuration(eventTime.end() , availableTime.end() - eventTime.end()));
+                   }
+                   else if(availableTime.end() > eventTime.end() && availableTime.start() < eventTime.end()){
+                       newAvailability.add(TimeRange.fromStartDuration(eventTime.end(), availableTime.end() - eventTime.end()));
+                   }
+                   else if(availableTime.end() > eventTime.start() && availableTime.start() <= eventTime.start()) {
+                       newAvailability.add(TimeRange.fromStartDuration(availableTime.start(), eventTime.start() - availableTime.start()));
+                   }
+                   else{
+                       newAvailability.add(availableTime);
+                   }
+               }
+               available = newAvailability;
+          }
+      }
+
+      //check if there is an available time available
+      Iterator<TimeRange> availableIterator = available.iterator();
+      Collection<TimeRange> newAvailability = new ArrayList<TimeRange>();
+      while (availableIterator.hasNext()){
+          TimeRange curTimeRange = availableIterator.next();
+          if(curTimeRange.duration() >= duration){
+              newAvailability.add(curTimeRange);
+          }
+      }
+    return newAvailability;
+  }
+}
```

Comment:
Nitpick: Github is complaining because files usually should end with an empty line.